### PR TITLE
Build PyOpenMS with numpy2

### LINKED ISSUE #7634: Upgrade numpy in pyopenms
Dear all,

the numpy version you are using for pyopenms https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/setup.py#L257 is quite outdated and incompatible with recent numpy versions on a binary level.

This forces me to downgrade or limit the numpy versions I'm able to use. This constraint may also restrict the Python versions which are compatible with pyopenms in the future, especially on systems without a pre-compiled installer.

The details about Numpy binary API stability are documented here: https://numpy.org/doc/1.22/user/depending_on_numpy.html

Thanks!

Uwe

### PR DESCRIPTION
**Pull request by poshul**
## Description

I've tested building with numpy2 instead of numpy1 and it looks like everything builds without any changes. This is intended to offer a longer term solution to https://github.com/OpenMS/OpenMS/issues/7511 .

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

___

### **PR Type**
enhancement, dependencies

___

### **Description**
- Updated the numpy version to the latest in the `.github/workflows/pyopenms-wheels.yml` file.
- Removed the specific version constraint (`numpy<=1.26.4`) and replaced it with `pip install -U numpy`.
- Ensures compatibility with newer versions of numpy for building PyOpenMS.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pyopenms-wheels.yml</strong><dd><code>Update numpy version to the latest in CI workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/pyopenms-wheels.yml

<li>Updated numpy version to the latest in multiple sections.<br> <li> Removed the specific version constraint for numpy.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7539/files#diff-084f20ae9727db0905c5fbbc32ef797e048c87887ccfc57b0f89b2bf82f45979">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by jpfeuffer** [comment]
I think setup.py needs to require numpy 2 as well then.

**Comment by jpfeuffer** [comment]
Ideally setup py would require the same major version it was built with.
Setup.py can and currently already is easily configured with cmake.

**Comment by poshul** [comment]
Yup. I should be able to add that. Honestly I didn't expect this to just
work out of the box.

On Wednesday, July 3, 2024, Julianus Pfeuffer ***@***.***>
wrote:
> Ideally setup py would require the same major version it was built with.
> Setup.py can and currently already is easily configured with cmake.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.<
https://ci3.googleusercontent.com/meips/ADKq_NZdLnI2eZowEqhmyyM4c3mhiFgJodQhhk70prcsiFm4QOd6PavRBHydake20Tad8R__rnCCx4yXyQTfDDG2DgWL41tEojyFM4vBw3EgTbA-saaSEkLS75gYIpeiMs4xTNRwV_evdmmYQbOADHkue4mwS2dHTSt_AqXA9NrmAAaD8zekvcWQwOvqJG79QuJPdCSXPEGsp2mNx6TM5tiJ_9YnD4aN8_017W2iXIbMubf0R1kFl1n674c=s0-d-e1-ft#https://github.com/notifications/beacon/AABD3C2JM3AK67JOV4FZMS3ZKQTJXA5CNFSM6AAAAABKJQOQNOWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTUDRDIQM.gif>Message
ID: ***@***.***>

**Comment by RalfG** [comment]
Hi @jpfeuffer and @poshul,

Thanks for looking in to this! I noticed the following comment:

> Ideally setup py would require the same major version it was built with.

However, with Numpy 2, its C-API is exposed in a backwards-compatible manner. So as long as pyOpenMS is built with Numpy 2, it will work on any installation with Numpy > 1.xx.
https://numpy.org/doc/2.1/dev/depending_on_numpy.html#numpy-2-0-specific-advice

So the install_requires Numpy dependency in setup.py can become `numpy>=1.23.5`

As an aside, are there plans for a pyOpenMS v3.2.1 soon?

Thanks!
Ralf

**Comment by jpfeuffer** [comment]
Ah great, I didn't know! Then definitely let's do >=1.23.5!

**Comment by poshul** [comment]
@jpfeuffer Done!
@RalfG If we need to we certainly can be. I want to see if the Visual C++ redist issue needs an update to the package.

**Comment by jpfeuffer** [comment]
I think he meant to install numpy 2 (or any latest) for the build but require numpy>=1.23.5 in the setup.py

File: .github/workflows/pyopenms-wheels.yml [hunk 1]
```diff
@@ -118,7 +118,7 @@ jobs:
           pip install -U pip
           pip install -U autowrap
           pip install -U pytest
-          pip install 'numpy<=1.26.4'
+          pip install -U numpy >=1.23.5
```
Comment (timosachsenberg) [inline .github/workflows/pyopenms-wheels.yml]: why 1.23.5 ?

**Review by jpfeuffer — APPROVED** [review]
Well windows is broken due to a faulty contrib release.. other than that, cannot see what could go wrong with those simple changes.

**Pull request merged**